src/gridcoin/contract/registry_db.h

```diff
@@ -0,0 +1,658 @@
+// Copyright (c) 2014-2023 The Gridcoin developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or https://opensource.org/licenses/mit-license.php.
+
+#ifndef GRIDCOIN_CONTRACT_REGISTRY_DB_H
+#define GRIDCOIN_CONTRACT_REGISTRY_DB_H
+
+#include "dbwrapper.h"
+#include "gridcoin/contract/handler.h"
+#include "gridcoin/contract/payload.h"
+#include "gridcoin/support/enumbytes.h"
+
+using LogFlags = BCLog::LogFlags;
+
+namespace GRC {
+//!
+//! \brief This is a template class generalization of the original beacon registry db in leveldb. This has been made into
+//! a template to facilitate usage of the db with strong typing for additional registries, but not have to repeat essentially
+//! the same code over and over. The intention will be to retrofit the beacon registry to use the template rather than
+//! the original code.
+//!
+//! The class template parameters are
+//! E: the entry type
+//! S: the entry status enum
+//! M: the map type for the entries
+//! H: the historical map type for historical entries
+//! R: the registry class itself (used for the first part of the key string name in leveldb
+//!
+template<class E, class S, class M, class H, class R>
+class RegistryDB
+{
+public:
+    //!
+    //! \brief The RegistryDB template constructor. The parameter is the version for the DB for the registry class.
+    //! \param version
+    //!
+    RegistryDB(uint32_t version)
+        : m_version(version)
+    {
+    };
+
+    //!
+    //! \brief Version number of the template entry db. This is dependent on the registry class instantiated, because
+    //! different registry classes were done at different times.
+    //!
+    //! CONSENSUS: Increment this value in the constructor of the registry class when introducing a breaking change
+    //! to the corrsponding entry class. This will ensure that when the wallet is restarted, the level db entry storage
```

Comment:
typo: "corresponding"

Reply:
Thanks. The linter doesn't catch the stuff in comments sometimes.